### DISCUSSION TITLE
Group-level session sharing, how to increase the number of parallel tasks

### DISCUSSION THREAD

**Discussion by shizhengchao** [apache/kyuubi#5155] [Q&A] [answered]
If ''spark.dynamicAllocation.maxExecutors'' is set to 400, then only 400 tasks can be run at the same time, and other newly submitted SQL will be blocked. How to solve this concurrency problem

**Comment by pan3793** [comment]
Increasing executor cores, typically, `spark.executor.cores` is set from 2 ~ 8

**Reply by shizhengchao** [reply]
What I mean is that as long as a job that takes up a lot of resources fills up the executor, other jobs can only wait for the resources to be released.

**Reply by pan3793** [reply, marked as answer]
https://spark.apache.org/docs/latest/job-scheduling.html#scheduling-within-an-application
